pandas/core/indexes/accessors.py

```diff
@@ -250,6 +250,32 @@ def isocalendar(self):
         """
         return self._get_values().isocalendar().set_index(self._parent.index)
 
+    @property
+    def weekofyear(self):
+        """
+        The week ordinal of the year.
+
+        .. deprecated:: 1.1.0
+
+        Series.dt.weekofyear and Series.dt.week have been deprecated.
+        Please use Series.dt.isocalendar().week instead.
+        """
+        import warnings
+
+        warnings.warn(
+            "Series.dt.weekofyear and Series.dt.week have been deprecated.  "
+            "Please use Series.dt.isocalendar().week instead.",
+            FutureWarning,
+            stacklevel=2,
+        )
+        week_series = self.isocalendar().week
+        week_series.name = self.name
+        if week_series.hasnans:
```

Comment:
we do it like this now?

Reply:
Yes, in the current `week` accessor the array is converted to a float and masked if there are nans.    This reproduces that behavior instead of passing back a `UInt32` series.